src/backend/core/tests/test_api_users.py

```diff
@@ -218,6 +218,80 @@ def test_api_users_authenticated_list_by_name_and_email():
     assert user_ids == [str(nicole.user.id), str(frank.user.id), str(david.user.id)]
 
 
+def test_api_users_authenticated_list_exclude_users_already_in_team(
+    django_assert_num_queries,
+):
+    """
+    Authenticated users should be able to search users
+    but the result should exclude all users already in the given team.
+    """
+    user = factories.UserFactory(email="tester@ministry.fr")
+    factories.IdentityFactory(user=user, email=user.email, name="john doe")
+    client = APIClient()
+    client.force_login(user)
+
+    dave = factories.IdentityFactory(name="dave bowman", email=None, is_main=True)
+    nicole = factories.IdentityFactory(name="nicole foole", email=None, is_main=True)
+    frank = factories.IdentityFactory(name="frank poole", email=None, is_main=True)
+    mary = factories.IdentityFactory(name="mary poole", email=None, is_main=True)
+    factories.IdentityFactory(name="heywood floyd", email=None, is_main=True)
+    factories.IdentityFactory(name="Andrei Smyslov", email=None, is_main=True)
+    factories.TeamFactory.create_batch(10)
+
+    # Add Dave and Frank in the same team
+    team = factories.TeamFactory(
+        users=[
+            (dave.user, models.RoleChoices.MEMBER),
+            (frank.user, models.RoleChoices.MEMBER),
+        ]
+    )
+    factories.TeamFactory(users=[(nicole.user, models.RoleChoices.MEMBER)])
+
+    # Search user to add him/her to a team, we give a team id to the request
+    # to exclude all users already in the team
+
+    # We can't find David Bowman because he is already a member of the given team
+    nb_queries = 1  # user authenticated
+    nb_queries += 1  # search user query
+    with django_assert_num_queries(nb_queries):
+        response = client.get(
+            f"/api/v1.0/users/?q=bowman&team_id={team.id}",
+        )
+    assert response.status_code == HTTP_200_OK
+    assert response.json()["results"] == []
+
+    # We can only find Nicole and Mary because Frank is already a member of the given team
+    nb_queries = 1  # user authenticated
+    nb_queries += 1  # search user query
+    nb_queries += 1  # User
+    nb_queries += 1  # Identity
+    with django_assert_num_queries(nb_queries):
+        response = client.get(
+            f"/api/v1.0/users/?q=ool&team_id={team.id}",
+        )
```

Comment:
nit-picking: we use a slightly different approach [here](https://github.com/numerique-gouv/people/blob/6d8e05b7464e1ec134078457744c4d6d20e29a2f/src/backend/core/tests/team_accesses/test_api_team_accesses_list.py#L183)